src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -456,6 +456,11 @@ static void window_editor_object_selection_mouseup(rct_window* w, rct_widgetinde
             {
                 finish_object_selection();
             }
+            if (gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER)
+            {
+                auto loadOrQuitAction = LoadOrQuitAction(LoadOrQuitModes::OpenSavePrompt, PromptMode::SaveBeforeQuit);
```

Comment:
This is a purely single player affair, so there is no need to do it through a game action. Additionally, there is nothing to save, so it should just exit to the main menu.

Reply:
I am very new to C++, but I will get better 🤞 
I think I've done it correctly now.